src/cli/cli_utils.hpp

```diff
@@ -0,0 +1,199 @@
+/*
+ *  Copyright (c) 2024, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file contains definitions of the CLI utils class.
+ */
+
+#ifndef CLI_UTILS_HPP_
+#define CLI_UTILS_HPP_
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+
+#include "cli/cli_output.hpp"
+
+#include "common/type_traits.hpp"
+
+namespace ot {
+namespace Cli {
+
+class CliUtils : public Output
+{
+public:
```

Comment:
Thinking more on this I have a suggestion:

We have `Output` as a common base class used by all CLI classes (we named it `Output` since its main job is to provide  `Output{Something}()` helper methods). In this PR we are adding a new `CliUtils` class as a new base class for CLI.

I suggest we combine these and have one base class.
- We can move these new `Get/Set` helper definition into the existing `Output` class (same file).
- We can then  rename it from `Output` to `Utils` (or some other name) to better reflect its broader scope.
- This provides a single base class offering all essential CLI helper methods.